adapters/redis/lib/opentelemetry/adapters/redis/patches/client.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Redis
+      module Patches
+        # Module to prepend to Redis::Client for instrumentation
+        module Client
+          def call(*args, &block)
```

Comment:
The gem has a couple of other `call_*` methods: `call_pipelined` (note the difference from `call_pipeline`) and `call_loop`. Have you considered patching `process` instead? The upside is that it is a bottleneck method that is called on all `call*` paths. The downside is that errors are sometimes raised outside its scope, so they're harder to intercept.

Reply:
@fbogsany I did not consider those because `dd-trace-rb` does not patch those methods.

Reply:
Relevant bits of Redis client: https://github.com/redis/redis-rb/blob/41395e9a1923fe72fad8403e60cb98627d515a85/lib/redis/client.rb#L124-L215

Reply:
I made an issue to track improvements: #197.